### DISCUSSION TITLE
API Batch Inference doesn't save labels / Slicer Server Error

### DISCUSSION THREAD

**Discussion by annkapopp** [Project-MONAI/MONAILabel#1587] [Q&A] [answered]
Hello!

I trained a DeepEdit model from scratch on a custom dataset. Now, I would like to execute the inference over the test images. I tried this by using  `/batch/infer/` on the API server but somehow, this doesn't work. The `/labels/original` folder is created but empty and the response body looks like this:

{
  "id": "c9ddb697-2a92-44ff-a375-7b32cf5ae1d2",
  "status": "SUBMITTED",
  "request": {
    "model": "deepedit_seg",
    "images": "all",
    "device": "cuda",
    "multi_gpu": true,
    "gpus": "all",
    "logging": "WARNING",
    "save_label": true,
    "label_tag": "original",
    "max_workers": 1,
    "max_batch_size": 1
  },
  "start_ts": "2023-11-06 12:33:48"
}

The model and the images are found though by executing `/datastore/` and  `/model/info/{model}`.
I started the server with:
`monailabel start_server --app MONAILABEL/apps/radiology --studies path_to_images --conf models deepedit --conf model_dir path_to_model --conf use_pretrained_model True --conf skip_trainers true
`

When I try to run inference in Slicer, there is an error "Failed to run inference in MONAI Label Server. Message:: Status 500; Response: Internal Server Error".

I hope that somebody can help me, thank you!

**Comment by annkapopp** [comment, marked as answer]
Solved it. I had to change the config in /radiology/lib/configs to add the path to my custom trained model.

**Comment by diazandr3s** [comment]
Hi @annkapopp,

Glad to see you manage to make it work.

Another way of doing batch inference is by running the main file in the radiology app folder: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/main.py#L267

Just provide these arguments: https://github.com/Project-MONAI/MONAILabel/blob/main/sample-apps/radiology/main.py#L295-L297

This will run MONAI Label without actually staring the server, just as any other python script.

Hope this helps,
